### PR TITLE
Monthly stats logs

### PR DESCRIPTION
**Pull request by Tehsurfer**
# Description

Create log for monthly stats
Reduce number of email sent in testing (We are almost always hitting the send grid max emails limit, making development difficult)

## Type of change

Delete those that don't apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Tehsurfer** [comment]
Can I get this merged in? I need this code in sparc-api so that the testing for sparc-api stops maxing out my send grid account

File: scripts/monthly_stats.py [hunk 1]
```diff
@@ -156,3 +172,18 @@ def send_email(self, email_address, email_body):
...
                                                                     email_body)
 
+    def send_logging_email(self, message):
+        status_code = 000
+        try:
+            status_code = self.send_grid.sendgrid_email_with_unsubscribe_group(Config.METRICS_EMAIL_ADDRESS,
+                                                                 Config.METRICS_EMAIL_ADDRESS,
+                                                                 'SPARC monthly dataset download summary',
+                                                                 message)
+            if status_code != 202:
```
Comment (hsorby) [inline scripts/monthly_stats.py:182]: Is there not a way to be more certain about the nature of the error?

File: tests/test_monthly_stats.py [hunk 2]
```diff
@@ -68,3 +68,7 @@ def test_email():  # Note that this will send an email to the test_"email_recipi
 def test_full_run():  # For each recipient, this will send an email to the test_email for each email that would have been sent to a user
     responses = ms.run()
     assert_true(False not in [r.status_code == 202 for r in responses])
+
+def test_log_email():
```
Comment (hsorby) [inline tests/test_monthly_stats.py:72]: Is `debug_mode` set for this test?

File: scripts/monthly_stats.py (hunk 1 above)
Comment (Tehsurfer) [reply scripts/monthly_stats.py:182]: Yeah we could, it sends 403 for if the rate limit is hit. 401 if the key is wrong. I can add that in

File: tests/test_monthly_stats.py (hunk 2 above)
Comment (Tehsurfer) [reply tests/test_monthly_stats.py:72]: Debug mode is set in creation of the monthly stats object at the top of this file

File: scripts/monthly_stats.py (hunk 1 above)
Comment (Tehsurfer) [reply scripts/monthly_stats.py]: Have added this in now

**Review by hsorby — CHANGES_REQUESTED** [review]

File: scripts/monthly_stats.py [hunk 3]
```diff
@@ -156,3 +172,22 @@ def send_email(self, email_address, email_body):
...
+            status_code = self.send_grid.sendgrid_email_with_unsubscribe_group(Config.METRICS_EMAIL_ADDRESS,
+                                                                 Config.METRICS_EMAIL_ADDRESS,
+                                                                 'SPARC monthly dataset download summary',
+                                                                 message)
+            if status_code == 403:
+                logging.error('Could not send sendgrid email because rate limit is hit (403)')
+            elif status_code == 401:
+                logging.error('Could not send sendgrid email. Sendgrid keys are likely incorrect (401)')
+            else:
+                logging.error(f'Unknown error. Status code: {status_code}')
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply scripts/monthly_stats.py:187]: Ooops yeah you are right

[the remark this answers is not in the document]
Comment (Tehsurfer) [reply scripts/monthly_stats.py]: Thanks @hsorby it is fixed now

**Review by hsorby — APPROVED** [review]
Looks good to me.

**Pull request merged**